### PR TITLE
Implemented Services in javascript

### PR DESCRIPTION
**Pull request by dvdthrowaway**
Here's my submission.

Let me know if you have any problems with installing the dependencies, instructions are in the readme.

### CODE REVIEW

**Comment by dvdthrowaway** [comment]
I just noticed the circleci dir, configuring now :)

File: src/ServiceMaster.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+const Services = [
+  require('./services/StravaService'),
+  require('./services/RwgpsService'),
+  require('./services/KomootService'),
+]
+let ServiceMap = {}
+
+for (let service of Services) {
+  ServiceMap[service.serviceName] = service
+}
```
Comment (brentnd) [inline src/ServiceMaster.js]: Can you modify `ServiceMaster` to follow the open-closed principle? (I'm not usually big on formal patterns/principles but this is the best way to say it without telling you what to do 😄).

Comment (dvdthrowaway) [reply src/ServiceMaster.js]: Sure! I think I see what you mean. The issue seems to me that to add another service, we’d have to modify service master directly. I think this means I’ll move master service away from a set of utility functions to a proper class that will be instanced and configured.  I’m traveling for work today but I’ll probably have some time tonight

Comment (brentnd) [reply src/ServiceMaster.js]: Cool. Seems like you understand.

**Comment by dvdthrowaway** [comment]
Sorry for the delayed response! Crazy times for me this week.

**Comment by dvdthrowaway** [comment]
as I was typing up a summary I realized that serviceMaster would make a good singleton.

**Comment by dvdthrowaway** [comment]
made some more changes, wanted to consolidate the configuration of service master and give it singleton.

**Comment by brentnd** [comment]
@dvdthrowaway thanks for the submission and the follow up to the review.

Closing this and @gooberholtzer will follow up with you.

**Pull request closed without merging**